services/motion/builtin/builtin.go

```diff
@@ -237,7 +258,103 @@ func (ms *builtIn) MoveOnGlobe(
 	angularVelocity float64,
 	extra map[string]interface{},
 ) (bool, error) {
-	return false, ErrNotImplemented
+	operation.CancelOtherWithLabel(ctx, builtinOpLabel)
+
+	// create a new empty framesystem which we add our base to
+	fs := referenceframe.NewEmptyFrameSystem("")
+
+	// get the localizer from the componentName
+	localizer, ok := ms.localizer[movementSensorName]
+	if !ok {
+		return false, resource.DependencyNotFoundError(movementSensorName)
+	}
+
+	currentPIF, _ := localizer.CurrentPosition(ctx)
+
+	// convert destination into spatialmath.Pose with respect to lat = 0 = lng
+	dstPose := spatialmath.GeoPointToPose(destination)
+
+	// convert the destination to be relative to the currentPIF
+	relativeDestinationPt := r3.Vector{
+		X: dstPose.Point().X - currentPIF.Pose().Point().X,
+		Y: dstPose.Point().Y - currentPIF.Pose().Point().Y,
+		Z: 0,
+	}
+
+	relativeDstPose := spatialmath.NewPoseFromPoint(relativeDestinationPt)
+	dstPIF := referenceframe.NewPoseInFrame(referenceframe.World, relativeDstPose)
+
+	// convert GeoObstacles into GeometriesInFrame with respect to the base's starting point
+	geoms := spatialmath.GeoObstaclesToGeometries(obstacles, currentPIF.Pose().Point())
+
+	gif := referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)
+	wrldst, err := referenceframe.NewWorldState([]*referenceframe.GeometriesInFrame{gif}, nil)
+	if err != nil {
+		return false, err
+	}
+
+	// construct limits
+	// TODO: have limits become a function of current position, destination and worldstate
```

Comment:
Do we have a ticket for this? This is approximately what we're doing for TP-space as well, but it's frankly pretty rudimentary. We should have a better way of defining "valid area", both automatically, and user-specified.

Reply:
Yeah lets remove the TODO and track this with a ticket if one doesn't exist already

Reply:
https://viam.atlassian.net/browse/RSDK-3558